### PR TITLE
Unit tests, bug fixes, code refactoring

### PR DESCRIPTION
**Pull request by fehlmach**
This change is [Reviewable](https://reviewable.io/reviews/netsec-ethz/rains/37)

### CODE REVIEW

**Comment by klausman** [comment]
Reviewed 77 of 77 files at r1.
Review status: all files reviewed at latest revision, 15 unresolved discussions.

---

*[proto/rainsMsg.capnp, line 7 at r1](https://reviewable.io:443/reviews/netsec-ethz/rains/37#-KoH4aRzizorRA_tiuoF:-KoH4aRzizorRA_tiuoG:b49ulyj) ([raw file](https://github.com/netsec-ethz/rains/blob/d51c7b14af2b6528862f49d522212fd1564578b0/proto/rainsMsg.capnp#L7)):*
> ```
> 
> struct RainsMessage  {
>     #RainsMessage contains the data of a message
> ```

This file has mixed tab/space indentation.

---

*[rainsd/serverInterfaceImpl.go, line 264 at r1](https://reviewable.io:443/reviews/netsec-ethz/rains/37#-KoHO5PdWgd6ueY8dUMv:-KoHO5PdWgd6ueY8dUMw:b-zh2kg4) ([raw file](https://github.com/netsec-ethz/rains/blob/d51c7b14af2b6528862f49d522212fd1564578b0/rainsd/serverInterfaceImpl.go#L264)):*
> ```Go
> func (c *pendingSignatureCacheImpl) GetAllAndDelete(context, zone string) ([]sectionWithSigSender, bool) {
> 	sections := []sectionWithSigSender{}
> 	deleteCount := uint(0)
> ```

Is there a reason why this has to be `unsigned`?

---

*[rainsd/switchboard.go, line 105 at r1](https://reviewable.io:443/reviews/netsec-ethz/rains/37#-KoHZd_c2a_DpOQUx4Pw:-KoHZd_c2a_DpOQUx4Px:b-ynv5nz) ([raw file](https://github.com/netsec-ethz/rains/blob/d51c7b14af2b6528862f49d522212fd1564578b0/rainsd/switchboard.go#L105)):*
> ```Go
> 			conn, err := listener.Accept()
> 			if err != nil {
> 				srvLogger.Error("lister could not accept connection", "error", err)
> ```

listener*

---

*[rainsdig/rainsdig.go, line 53 at r1](https://reviewable.io:443/reviews/netsec-ethz/rains/37#-KoH_y_VahczYgVfhzQT:-KoH_y_VahczYgVfhzQU:b30fipb) ([raw file](https://github.com/netsec-ethz/rains/blob/d51c7b14af2b6528862f49d522212fd1564578b0/rainsdig/rainsdig.go#L53)):*
> ```Go
> 	7: Disable verification delegation (client protocol only)
> 	8: Suppress proactive caching of future assertions
> 	e.g. to specify query options 4 and 2 with higher priority on option 4 write: -qopt=4 -qopt=2
> ```

Add a TODO that this list should be generated from internal constants.

---

*[rainslib/datamodel_test.go, line 1 at r1](https://reviewable.io:443/reviews/netsec-ethz/rains/37#-KoHcUJ6vDNuf7wIH-h3:-KoHcUJ6vDNuf7wIH-h4:b9xuy6e) ([raw file](https://github.com/netsec-ethz/rains/blob/d51c7b14af2b6528862f49d522212fd1564578b0/rainslib/datamodel_test.go#L1)):*
> ```Go
> package rainslib
> ```

I can't run these tests myself, since I don't have `test/test.gob`, which should be in the repo (but possibly in a `testdata/` folder

---

*[rainslib/objects.go, line 9 at r1](https://reviewable.io:443/reviews/netsec-ethz/rains/37#-KoHdrZ0WpNTzrlzDlck:-KoHdrZ0WpNTzrlzDlcl:b-bma7jg) ([raw file](https://github.com/netsec-ethz/rains/blob/d51c7b14af2b6528862f49d522212fd1564578b0/rainslib/objects.go#L9)):*
> ```Go
> 	"strconv"
> 
> 	"encoding/hex"
> ```

Remove empty line above this (std lib imports should always be one block).

---

*[rainslib/rainslibUtil_test.go, line 133 at r1](https://reviewable.io:443/reviews/netsec-ethz/rains/37#-KoHeU0FCg2gF9128v99:-KoHeU0FCg2gF9128v9A:b-dezi4w) ([raw file](https://github.com/netsec-ethz/rains/blob/d51c7b14af2b6528862f49d522212fd1564578b0/rainslib/rainslibUtil_test.go#L133)):*
> ```Go
> 	token := GenerateToken()
> 	_, subjectAddress1, _ := net.ParseCIDR("127.0.0.1/32")
> 	_, subjectAddress2, _ := net.ParseCIDR(ip6TestAddrCIDR)
> ```

Why is the ipv6 test addr a var/constant and the v4 one a literal?

---

*[rainspub/test/server.conf, line 3 at r1](https://reviewable.io:443/reviews/netsec-ethz/rains/37#-KoHgKob5xL1p-V7qgg3:-KoHgKob5xL1p-V7qgg4:blpcpt7) ([raw file](https://github.com/netsec-ethz/rains/blob/d51c7b14af2b6528862f49d522212fd1564578b0/rainspub/test/server.conf#L3)):*
> ```
> {
>     "RootZonePublicKeyPath":        "test/selfSignedRootDelegationAssertion.gob",
>     "ServerAddress":                {
> ```

This file has mixed space/tab indentation.

---

*[rainsSiglib/rainsSiglib.go, line 8 at r1](https://reviewable.io:443/reviews/netsec-ethz/rains/37#-KoHd2gu47Er-QIDIx_z:-KoHd2gu47Er-QIDIxa-:b-yf3mfn) ([raw file](https://github.com/netsec-ethz/rains/blob/d51c7b14af2b6528862f49d522212fd1564578b0/rainsSiglib/rainsSiglib.go#L8)):*
> ```Go
> import (
> 	"fmt"
> 	"rains/rainslib"
> ```

Do not use relative imports like this. This should be `github.com/netsec-ethz/rains/rainslib` and moved below the log import below. The same for all other imports from rains itself (elsewhere).

Basically, only stdlib imports should be relative/bare (`"foo"` or `"bar/baz"`), everything else should be along the "`github.com/user/repo"` scheme.

---

*[serverCH/config/server.conf, line 20 at r1](https://reviewable.io:443/reviews/netsec-ethz/rains/37#-KoHgwlRwqKUNOt0zshI:-KoHgwlRwqKUNOt0zshJ:b-gosqs1) ([raw file](https://github.com/netsec-ethz/rains/blob/d51c7b14af2b6528862f49d522212fd1564578b0/serverCH/config/server.conf#L20)):*
> ```
>     "NormalBufferSize":             100000,
>     "PrioWorkerCount":              2,
> 	"NormalWorkerCount":            10,
> ```

Mixed indentation.

---

*[serverCH/zoneFiles/chZoneDelegation.txt, line 5 at r1](https://reviewable.io:443/reviews/netsec-ethz/rains/37#-KoHh0-daK5oH9S0enaE:-KoHh0-daK5oH9S0enaF:b-2b4p4j) ([raw file](https://github.com/netsec-ethz/rains/blob/d51c7b14af2b6528862f49d522212fd1564578b0/serverCH/zoneFiles/chZoneDelegation.txt#L5)):*
> ```
>                         :A: ch [ :deleg: ed25519 2245137bcb058f799843bb8c6df31927b547e4951142b99ae97c668b076e9d84 ]
>                     ]
>             ]
> ```

Missing CR at EOF.

---

*[serverCH/zoneFiles/chZoneFile.txt, line 6 at r1](https://reviewable.io:443/reviews/netsec-ethz/rains/37#-KoHh66oAHd1ijoA-6Y3:-KoHh66oAHd1ijoA-6Y4:bu5v99c) ([raw file](https://github.com/netsec-ethz/rains/blob/d51c7b14af2b6528862f49d522212fd1564578b0/serverCH/zoneFiles/chZoneFile.txt#L6)):*
> ```
>                         :A: ethz [ :ip4: 129.132.128.139 ]
>                     ]
>             ]
> ```

Missing CR.

---

*[serverRoot/config/server.conf, line 1 at r1](https://reviewable.io:443/reviews/netsec-ethz/rains/37#-KoHhAwpkUL6GxSX02_e:-KoHhAwpkUL6GxSX02_f:b-gosqs1) ([raw file](https://github.com/netsec-ethz/rains/blob/d51c7b14af2b6528862f49d522212fd1564578b0/serverRoot/config/server.conf#L1)):*
> ```
> {
> ```

Mixed indentation.

---

*[serverRoot/zoneFiles/rootZoneFile.txt, line 5 at r1](https://reviewable.io:443/reviews/netsec-ethz/rains/37#-KoHhDo86ZtR0wXem1on:-KoHhDo86ZtR0wXem1oo:b7uhnpq) ([raw file](https://github.com/netsec-ethz/rains/blob/d51c7b14af2b6528862f49d522212fd1564578b0/serverRoot/zoneFiles/rootZoneFile.txt#L5)):*
> ```
>                         :A: ch [ :deleg: ed25519 2245137bcb058f799843bb8c6df31927b547e4951142b99ae97c668b076e9d84 ]
>                     ]
>             ]
> ```

Missing CR

---

*[utils/protoParser/protoEncoder.go, line 11 at r1](https://reviewable.io:443/reviews/netsec-ethz/rains/37#-KoHhPk-Q3VRJA8-XEBa:-KoHhPk-Q3VRJA8-XEBb:b-fvvuqs) ([raw file](https://github.com/netsec-ethz/rains/blob/d51c7b14af2b6528862f49d522212fd1564578b0/utils/protoParser/protoEncoder.go#L11)):*
> ```Go
> 
> 	log "github.com/inconshreveable/log15"
> 	"golang.org/x/crypto/ed25519"
> ```

Insert empty line above (renamed third-party imports should be a separate block between stdlib and non-renamed 3rd party imports).

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/netsec-ethz/rains/37)*

**Comment by klausman** [comment]
Reviewed 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 16 unresolved discussions.

---

*[DESIGN-SIGNATURES.md, line 3 at r2](https://reviewable.io:443/reviews/netsec-ethz/rains/37#-KoIQo4YGjqxNkbIjtfA:-KoIQo4YGjqxNkbIjtfB:b-kkzuo6) ([raw file](https://github.com/netsec-ethz/rains/blob/f621043721215f7e718831ec2adfa246329902a1/DESIGN-SIGNATURES.md#L3)):*
> ```Markdown
> # Signature caching and lookup design
> 
> ##requirements for section signatures
> ```

Please reflow this to 80 (or 100) characters per line, since editing is easier that way. the MD-to-whatever converters will do the right thing.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/netsec-ethz/rains/37)*

**Comment by klausman** [comment]
Review status: all files reviewed at latest revision, 17 unresolved discussions.

---

*[DESIGN-SIGNATURES.md, line 43 at r2](https://reviewable.io:443/reviews/netsec-ethz/rains/37#-KoIR7NzOTx3Y7IVGnT7:-KoIR7NzOTx3Y7IVGnT8:bu5v99c) ([raw file](https://github.com/netsec-ethz/rains/blob/f621043721215f7e718831ec2adfa246329902a1/DESIGN-SIGNATURES.md#L43)):*
> ```Markdown
> ##conclusion
> The second proposal is a generalization of the first proposal (i.e. that the second proposal can be used by an authority as the first one but also gives more freedom to abuse it). As long as there is no key revocation mechanism the second proposal only adds performance overhead. But if in a later stage a key revocation mechanism is added the second approach might yield more security as then a single key compromise affects the system only minimally (until the revocation is pushed). But this approach gives an incentive to add more signatures to a section which then influences the performance of the overall system negatively. It is also less important for an authority to really keep a private key safe, as there is a backup. Also it is not really clear how an authority can discover a key compromise in a timely fashion and then issue a revocation which also takes some time to spread over the network.
> If we want to make sure that an authority really cares about keeping its private keys safe then we should choose the first proposal. It also gives an incentive to authorities to choose a short validity live time for public keys such that a compromised key can only be used in a short amount of time. Additionally, it has a positive performance impact as less signatures are used per section, cache lookups are faster and the sections are slightly smaller.
> ```

Missing CR.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/netsec-ethz/rains/37)*

**Comment by fehlmach** [comment]
Review status: all files reviewed at latest revision, 17 unresolved discussions.

---

*[DESIGN-SIGNATURES.md, line 3 at r2](https://reviewable.io:443/reviews/netsec-ethz/rains/37#-KoIQo4YGjqxNkbIjtfA:-KoIXiYEWppYWQqYbkCK:b-896fix) ([raw file](https://github.com/netsec-ethz/rains/blob/f621043721215f7e718831ec2adfa246329902a1/DESIGN-SIGNATURES.md#L3)):*
<details><summary><i>Previously, klausman (Tobias Klausmann) wrote…</i></summary><blockquote>

Please reflow this to 80 (or 100) characters per line, since editing is easier that way. the MD-to-whatever converters will do the right thing.
</blockquote></details>

Done.

---

*[DESIGN-SIGNATURES.md, line 43 at r2](https://reviewable.io:443/reviews/netsec-ethz/rains/37#-KoIR7NzOTx3Y7IVGnT7:-KoIXm0GtVClUozUFj78:b-896fix) ([raw file](https://github.com/netsec-ethz/rains/blob/f621043721215f7e718831ec2adfa246329902a1/DESIGN-SIGNATURES.md#L43)):*
<details><summary><i>Previously, klausman (Tobias Klausmann) wrote…</i></summary><blockquote>

Missing CR.
</blockquote></details>

Done.

---

*[proto/rainsMsg.capnp, line 7 at r1](https://reviewable.io:443/reviews/netsec-ethz/rains/37#-KoH4aRzizorRA_tiuoF:-KoIYDtoarbYF8fs0pPQ:b-896fix) ([raw file](https://github.com/netsec-ethz/rains/blob/d51c7b14af2b6528862f49d522212fd1564578b0/proto/rainsMsg.capnp#L7)):*
<details><summary><i>Previously, klausman (Tobias Klausmann) wrote…</i></summary><blockquote>

This file has mixed tab/space indentation.
</blockquote></details>

Done.

---

*[rainsd/switchboard.go, line 105 at r1](https://reviewable.io:443/reviews/netsec-ethz/rains/37#-KoHZd_c2a_DpOQUx4Pw:-KoIYLCA3D3160KlNIfV:b-896fix) ([raw file](https://github.com/netsec-ethz/rains/blob/d51c7b14af2b6528862f49d522212fd1564578b0/rainsd/switchboard.go#L105)):*
<details><summary><i>Previously, klausman (Tobias Klausmann) wrote…</i></summary><blockquote>

listener*
</blockquote></details>

Done.

---

*[rainsdig/rainsdig.go, line 53 at r1](https://reviewable.io:443/reviews/netsec-ethz/rains/37#-KoH_y_VahczYgVfhzQT:-KoIYYoPGaqg6u2iN0G8:b-896fix) ([raw file](https://github.com/netsec-ethz/rains/blob/d51c7b14af2b6528862f49d522212fd1564578b0/rainsdig/rainsdig.go#L53)):*
<details><summary><i>Previously, klausman (Tobias Klausmann) wrote…</i></summary><blockquote>

Add a TODO that this list should be generated from internal constants.
</blockquote></details>

Done.

---

*[rainslib/datamodel_test.go, line 1 at r1](https://reviewable.io:443/reviews/netsec-ethz/rains/37#-KoHcUJ6vDNuf7wIH-h3:-KoIYm0X5Tn-zCbYnl-b:b-896fix) ([raw file](https://github.com/netsec-ethz/rains/blob/d51c7b14af2b6528862f49d522212fd1564578b0/rainslib/datamodel_test.go#L1)):*
<details><summary><i>Previously, klausman (Tobias Klausmann) wrote…</i></summary><blockquote>

I can't run these tests myself, since I don't have `test/test.gob`, which should be in the repo (but possibly in a `testdata/` folder
</blockquote></details>

Done.

---

*[rainslib/objects.go, line 9 at r1](https://reviewable.io:443/reviews/netsec-ethz/rains/37#-KoHdrZ0WpNTzrlzDlck:-KoIYuMzlRLAtLpmbQan:b-896fix) ([raw file](https://github.com/netsec-ethz/rains/blob/d51c7b14af2b6528862f49d522212fd1564578b0/rainslib/objects.go#L9)):*
<details><summary><i>Previously, klausman (Tobias Klausmann) wrote…</i></summary><blockquote>

Remove empty line above this (std lib imports should always be one block).
</blockquote></details>

Done.

---

*[rainslib/rainslibUtil_test.go, line 133 at r1](https://reviewable.io:443/reviews/netsec-ethz/rains/37#-KoHeU0FCg2gF9128v99:-KoI_2THRHe7qHhtTGl0:b-nk1tgg) ([raw file](https://github.com/netsec-ethz/rains/blob/d51c7b14af2b6528862f49d522212fd1564578b0/rainslib/rainslibUtil_test.go#L133)):*
<details><summary><i>Previously, klausman (Tobias Klausmann) wrote…</i></summary><blockquote>

Why is the ipv6 test addr a var/constant and the v4 one a literal?
</blockquote></details>

wrong line of thought on my part. fixed

---

*[rainspub/test/server.conf, line 3 at r1](https://reviewable.io:443/reviews/netsec-ethz/rains/37#-KoHgKob5xL1p-V7qgg3:-KoI_GeDLpR-L1Y4RGAR:b-896fix) ([raw file](https://github.com/netsec-ethz/rains/blob/d51c7b14af2b6528862f49d522212fd1564578b0/rainspub/test/server.conf#L3)):*
<details><summary><i>Previously, klausman (Tobias Klausmann) wrote…</i></summary><blockquote>

This file has mixed space/tab indentation.
</blockquote></details>

Done.

---

*[serverCH/config/server.conf, line 20 at r1](https://reviewable.io:443/reviews/netsec-ethz/rains/37#-KoHgwlRwqKUNOt0zshI:-KoI_QZmUQcJHqpT74GF:b-896fix) ([raw file](https://github.com/netsec-ethz/rains/blob/d51c7b14af2b6528862f49d522212fd1564578b0/serverCH/config/server.conf#L20)):*
<details><summary><i>Previously, klausman (Tobias Klausmann) wrote…</i></summary><blockquote>

Mixed indentation.
</blockquote></details>

Done.

---

*[serverCH/zoneFiles/chZoneDelegation.txt, line 5 at r1](https://reviewable.io:443/reviews/netsec-ethz/rains/37#-KoHh0-daK5oH9S0enaE:-KoI_Z9oUF-_ujD0ZjMJ:b-896fix) ([raw file](https://github.com/netsec-ethz/rains/blob/d51c7b14af2b6528862f49d522212fd1564578b0/serverCH/zoneFiles/chZoneDelegation.txt#L5)):*
<details><summary><i>Previously, klausman (Tobias Klausmann) wrote…</i></summary><blockquote>

Missing CR at EOF.
</blockquote></details>

Done.

---

*[serverCH/zoneFiles/chZoneFile.txt, line 6 at r1](https://reviewable.io:443/reviews/netsec-ethz/rains/37#-KoHh66oAHd1ijoA-6Y3:-KoI_Zw1j2hIftvOzbrR:b-896fix) ([raw file](https://github.com/netsec-ethz/rains/blob/d51c7b14af2b6528862f49d522212fd1564578b0/serverCH/zoneFiles/chZoneFile.txt#L6)):*
<details><summary><i>Previously, klausman (Tobias Klausmann) wrote…</i></summary><blockquote>

Missing CR.
</blockquote></details>

Done.

---

*[serverRoot/config/server.conf, line 1 at r1](https://reviewable.io:443/reviews/netsec-ethz/rains/37#-KoHhAwpkUL6GxSX02_e:-KoI_l7gO4WV5OkKliAw:b-896fix) ([raw file](https://github.com/netsec-ethz/rains/blob/d51c7b14af2b6528862f49d522212fd1564578b0/serverRoot/config/server.conf#L1)):*
<details><summary><i>Previously, klausman (Tobias Klausmann) wrote…</i></summary><blockquote>

Mixed indentation.
</blockquote></details>

Done.

---

*[serverRoot/zoneFiles/rootZoneFile.txt, line 5 at r1](https://reviewable.io:443/reviews/netsec-ethz/rains/37#-KoHhDo86ZtR0wXem1on:-KoI_nG-G00607x9CgLV:b-896fix) ([raw file](https://github.com/netsec-ethz/rains/blob/d51c7b14af2b6528862f49d522212fd1564578b0/serverRoot/zoneFiles/rootZoneFile.txt#L5)):*
<details><summary><i>Previously, klausman (Tobias Klausmann) wrote…</i></summary><blockquote>

Missing CR
</blockquote></details>

Done.

---

*[utils/protoParser/protoEncoder.go, line 11 at r1](https://reviewable.io:443/reviews/netsec-ethz/rains/37#-KoHhPk-Q3VRJA8-XEBa:-KoI_w6-CxM8rN7dLgdu:b-896fix) ([raw file](https://github.com/netsec-ethz/rains/blob/d51c7b14af2b6528862f49d522212fd1564578b0/utils/protoParser/protoEncoder.go#L11)):*
<details><summary><i>Previously, klausman (Tobias Klausmann) wrote…</i></summary><blockquote>

Insert empty line above (renamed third-party imports should be a separate block between stdlib and non-renamed 3rd party imports).
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/netsec-ethz/rains/37)*

**Comment by fehlmach** [comment]
Review status: 57 of 75 files reviewed at latest revision, 17 unresolved discussions.

---

*[rainsd/serverInterfaceImpl.go, line 264 at r1](https://reviewable.io:443/reviews/netsec-ethz/rains/37#-KoHO5PdWgd6ueY8dUMv:-KoLi0wZ2W90Hwkkp5jJ:bv0k0ql) ([raw file](https://github.com/netsec-ethz/rains/blob/d51c7b14af2b6528862f49d522212fd1564578b0/rainsd/serverInterfaceImpl.go#L264)):*
<details><summary><i>Previously, klausman (Tobias Klausmann) wrote…</i></summary><blockquote>

Is there a reason why this has to be `unsigned`?
</blockquote></details>

because the field in the struct counting the elements is a uint. Should I change it to int? It would remove all those type casts.

---

*[rainsSiglib/rainsSiglib.go, line 8 at r1](https://reviewable.io:443/reviews/netsec-ethz/rains/37#-KoHd2gu47Er-QIDIx_z:-KoLigivO-7etHM2d5h1:b-vnjqp2) ([raw file](https://github.com/netsec-ethz/rains/blob/d51c7b14af2b6528862f49d522212fd1564578b0/rainsSiglib/rainsSiglib.go#L8)):*
<details><summary><i>Previously, klausman (Tobias Klausmann) wrote…</i></summary><blockquote>

Do not use relative imports like this. This should be `github.com/netsec-ethz/rains/rainslib` and moved below the log import below. The same for all other imports from rains itself (elsewhere).

Basically, only stdlib imports should be relative/bare (`"foo"` or `"bar/baz"`), everything else should be along the "`github.com/user/repo"` scheme.
</blockquote></details>

Yes that should be done for the netsec-ethz repro, but I work on a fork, where it then wold be github.com/fehlmach/rains/rainslib which also does not match the above description. How can I resolve that?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/netsec-ethz/rains/37)*

**Comment by fehlmach** [comment]
Review status: 23 of 98 files reviewed at latest revision, 17 unresolved discussions.

---

*[rainsSiglib/rainsSiglib.go, line 8 at r1](https://reviewable.io:443/reviews/netsec-ethz/rains/37#-KoHd2gu47Er-QIDIx_z:-KoRNjpGQeKoZkboYwI1:bb7f2mf) ([raw file](https://github.com/netsec-ethz/rains/blob/d51c7b14af2b6528862f49d522212fd1564578b0/rainsSiglib/rainsSiglib.go#L8)):*
<details><summary><i>Previously, fehlmach wrote…</i></summary><blockquote>

Yes that should be done for the netsec-ethz repro, but I work on a fork, where it then wold be github.com/fehlmach/rains/rainslib which also does not match the above description. How can I resolve that?
</blockquote></details>

done

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/netsec-ethz/rains/37)*

**Comment by britram** [comment]
Reviewed 27 of 81 files at r1, 79 of 80 files at r3.
Review status: all files reviewed at latest revision, 21 unresolved discussions.

---

*[rainsd/serverInterfaceImpl.go, line 264 at r1](https://reviewable.io:443/reviews/netsec-ethz/rains/37#-KoHO5PdWgd6ueY8dUMv:-KoROt55SZeYmJ9K0-s4:b-itzdtq) ([raw file](https://github.com/netsec-ethz/rains/blob/d51c7b14af2b6528862f49d522212fd1564578b0/rainsd/serverInterfaceImpl.go#L264)):*
<details><summary><i>Previously, fehlmach wrote…</i></summary><blockquote>

because the field in the struct counting the elements is a uint. Should I change it to int? It would remove all those type casts.
</blockquote></details>

not really a reason; do switch to int.

---

*[rainsd/verify.go, line 37 at r3](https://reviewable.io:443/reviews/netsec-ethz/rains/37#-KoRPKt1qKTc2nh2W-Jo:-KoRPKt1qKTc2nh2W-Jp:be73jqn) ([raw file](https://github.com/netsec-ethz/rains/blob/ed59448111520f6edce40f6baebb74e163669618/rainsd/verify.go#L37)):*
> ```Go
> 	//init cache
> 	var err error
> 	//FIXME CFE rethink design of this cache! Should it be possible that two public keys are valid at the same time? if so then it must be specified which one is used
> ```

yes, per our design discussion yesterday.

---

*[rainsdig/rainsdig.go, line 53 at r1](https://reviewable.io:443/reviews/netsec-ethz/rains/37#-KoH_y_VahczYgVfhzQT:-KoRPWOw4THVxg7WDltu:b-4b52uu) ([raw file](https://github.com/netsec-ethz/rains/blob/d51c7b14af2b6528862f49d522212fd1564578b0/rainsdig/rainsdig.go#L53)):*
<details><summary><i>Previously, fehlmach wrote…</i></summary><blockquote>

Done.
</blockquote></details>

eventually, I'd like string mnenomics for these on the command-line too; please file an issue.

---

*[rainsdig/rainsdig.go, line 188 at r3](https://reviewable.io:443/reviews/netsec-ethz/rains/37#-KoRPgaG7u3Q5hedZWJ6:-KoRPgaG7u3Q5hedZWJ7:b-qlyr70) ([raw file](https://github.com/netsec-ethz/rains/blob/ed59448111520f6edce40f6baebb74e163669618/rainsdig/rainsdig.go#L188)):*
> ```Go
> 
> //Set transforms command line input of a query option to its internal representation
> func (i *qoptFlag) Set(value string) error {
> ```

...and this would be the place to make the string mnemonics happen...

---

*[serverRoot/rainsServer.go, line 1 at r3](https://reviewable.io:443/reviews/netsec-ethz/rains/37#-KoRT42h7WxsOc8TRuIR:-KoRT42iEMVtgdn9wMaa:bf36h1b) ([raw file](https://github.com/netsec-ethz/rains/blob/ed59448111520f6edce40f6baebb74e163669618/serverRoot/rainsServer.go#L1)):*
> ```Go
> package main
> ```

Obviously we'll want a test harness (/ rainsd CLI) for the future that allows us to bring a different server instance up without writing Go code.

---

*[utils/cache/cache.go, line 16 at r3](https://reviewable.io:443/reviews/netsec-ethz/rains/37#-KoRSVmj5VZO1q01I-92:-KoRSVmkm1RdBS-aEBZO:bb7i8xh) ([raw file](https://github.com/netsec-ethz/rains/blob/ed59448111520f6edce40f6baebb74e163669618/utils/cache/cache.go#L16)):*
> ```Go
> type Cache struct {
> 	//maxEntries is the maximum number of cache entries before an item is evicted.
> 	maxEntries uint
> ```

int?

---

*[serverRoot/zoneFiles/rootZoneFile.txt, line 5 at r1](https://reviewable.io:443/reviews/netsec-ethz/rains/37#-KoHhDo86ZtR0wXem1on:-KoRS6n5NdXXQ38AvwEj:b-sizqu2) ([raw file](https://github.com/netsec-ethz/rains/blob/d51c7b14af2b6528862f49d522212fd1564578b0/serverRoot/zoneFiles/rootZoneFile.txt#L5)):*
<details><summary><i>Previously, fehlmach wrote…</i></summary><blockquote>

Done.
</blockquote></details>

wait, why is this empty?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/netsec-ethz/rains/37)*

**Comment by britram** [comment]
:lgtm:

---

Review status: all files reviewed at latest revision, 18 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/netsec-ethz/rains/37#-:-KoRTXgOP5-X0jm7N2Mk:bnfp4nl)*

**Comment by britram** [comment]
Reviewed 1 of 80 files at r3.
Review status: all files reviewed at latest revision, 18 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/netsec-ethz/rains/37)*

**Comment by fehlmach** [comment]
Review status: all files reviewed at latest revision, 19 unresolved discussions.

---

*[rainsdig/rainsdig.go, line 53 at r1](https://reviewable.io:443/reviews/netsec-ethz/rains/37#-KoH_y_VahczYgVfhzQT:-KoRVQ59Pagw0PU8asYC:b-896fix) ([raw file](https://github.com/netsec-ethz/rains/blob/d51c7b14af2b6528862f49d522212fd1564578b0/rainsdig/rainsdig.go#L53)):*
<details><summary><i>Previously, britram (Brian Trammell) wrote…</i></summary><blockquote>

eventually, I'd like string mnenomics for these on the command-line too; please file an issue.
</blockquote></details>

Done.

---

*[serverRoot/rainsServer.go, line 1 at r3](https://reviewable.io:443/reviews/netsec-ethz/rains/37#-KoRT42h7WxsOc8TRuIR:-KoRVf7f7aDVAx85X_Zk:b2nhswv) ([raw file](https://github.com/netsec-ethz/rains/blob/ed59448111520f6edce40f6baebb74e163669618/serverRoot/rainsServer.go#L1)):*
<details><summary><i>Previously, britram (Brian Trammell) wrote…</i></summary><blockquote>

Obviously we'll want a test harness (/ rainsd CLI) for the future that allows us to bring a different server instance up without writing Go code.
</blockquote></details>

Yep

---

*[utils/cache/cache.go, line 16 at r3](https://reviewable.io:443/reviews/netsec-ethz/rains/37#-KoRSVmj5VZO1q01I-92:-KoRVhL_ykPxWiai1Eod:b-s3ozd0) ([raw file](https://github.com/netsec-ethz/rains/blob/ed59448111520f6edce40f6baebb74e163669618/utils/cache/cache.go#L16)):*
<details><summary><i>Previously, britram (Brian Trammell) wrote…</i></summary><blockquote>

int?
</blockquote></details>

Yep, I will change that during refactoring of the cache

---

*[serverRoot/zoneFiles/rootZoneFile.txt, line 5 at r1](https://reviewable.io:443/reviews/netsec-ethz/rains/37#-KoHhDo86ZtR0wXem1on:-KoRW4iprD92VT_LT-UM:b-i09pnw) ([raw file](https://github.com/netsec-ethz/rains/blob/d51c7b14af2b6528862f49d522212fd1564578b0/serverRoot/zoneFiles/rootZoneFile.txt#L5)):*
<details><summary><i>Previously, britram (Brian Trammell) wrote…</i></summary><blockquote>

wait, why is this empty?
</blockquote></details>

file moved

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/netsec-ethz/rains/37)*

**Comment by klausman** [comment]
:lgtm:

---

Reviewed 2 of 81 files at r1, 80 of 80 files at r3.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/netsec-ethz/rains/37#-:-KoReLQmevJ-CFnf_Nl8:bnfp4nl)*
